docs/source/developing/ci.adoc

```diff
@@ -10,21 +10,45 @@ toc::[]
 [[ci-overview]]
 == Overview
 
-The following sections describes the various CI setups Minishift utilizes.
+Minishift uses the following CI build servers: CentOS, AppVeyor, Circle, and Travis. These servers build incoming pull requests and provide the status of the builds to ensure continuous integration. Except for Travis CI, they also archive build artifacts of successful builds.
 
-[[ci-setup]]
+image::<CI_Servers>[CI Servers]
+
+The following sections describe the various CI build servers Minishift utilizes.
+
+[[centos-ci]]
 == CentOS CI
 
-Minishift uses several CI build servers, including https://ci.centos.org/[CentOS CI]. The server
-builds incoming pull requests and pushes to master. It also archives build artifacts of successful
-builds.
+https://ci.centos.org/[CentOS CI] server is the only CI server which can run integration tests, as it supports nested virtualization. Thus it is used to run integration tests as well as unit tests and code format tests. It also verifies the cross compilation of the binary.
 
-You can find the CentOS CI Jenkins job for Minishift https://ci.centos.org/job/minishift/[here].
+You can find the CentOS CI Jenkins job for Minishift https://ci.centos.org/job/minishift/[here]. Refer to the https://github.com/minishift/minishift/blob/master/centos_ci.sh[CentOS CI config] file to see its configuration.
 
 The artifacts of a successful pull request build can be found at
```

Comment:
IMO this whole sections renders really poorly. It is just a bunch of single sentences, each grouped as paragraph. Either we write a cohesive paragraph and pull this together or we use a list of forms.